### ISSUE TITLE
Fonts of text in a mixed-mode L-string differs from a normal text

### ISSUE THREAD

**Issue by ShuhuaGao** [JuliaPlots/Makie.jl#1560]
```julia
using CairoMakie
using LaTeXStrings

fig = Figure(resolution=(600, 400))
ax = Axis(fig[1, 1], xlabel=L"Time $t$", xlabelsize=22, ylabel="Price", ylabelsize=22)
lines!(ax, 1:24, rand(24))
fig
```
I expect that the font of "Time" is the same as "Price". However, it does not.
![pt](https://user-images.githubusercontent.com/20141984/148668295-056eaad3-4d5c-40bd-84b8-cda991590e65.png)

What should I do to make Makie render them with the same font (better by default)?

**Comment by jonas-schulze** [comment]
Regular strings like `"Price"` and all the tick labels are rendered using the font configured in Makie, while `LaTeXString`s like `L"Time $t$"` are rendered by [MathTeXEngine.jl](https://github.com/Kolaru/MathTeXEngine.jl) using (New) Computer Modern.

I'm currently working on getting a different font working in MathTeXEngine, but with no success so far. The documentation of MathTeXEngine says

> Currently the only font set supported is Computer Modern.

but there clearly is some modularity prepared to support other fonts. Here is what I did so far:

```julia
utopia_dir = "/usr/local/texlive/2018/texmf-dist/fonts/type1/adobe/utopia"

utopia = Dict(
	:regular => joinpath(utopia_dir, "putr8a.pfb"),
	:italic => joinpath(utopia_dir, "putri8a.pfb"),
	:bold => joinpath(utopia_dir, "putb8a.pfb"),
	:bolditalic => joinpath(utopia_dir, "putbi8a.pfb"),
	:math => joinpath(utopia_dir, "putri8a.pfb"), # italic
)

utopia_ff = Makie.MathTeXEngine.FontFamily(utopia)

Makie.MathTeXEngine.generate_tex_elements("a+b", utopia_ff)
```

Currently, the values of the `utopia` dictionary must be strings to support the current [`get_font`](https://github.com/Kolaru/MathTeXEngine.jl/blob/f07a5dfab247e9144dfc1076ab13a1badc14c156/src/engine/fonts.jl#L84-L92) and [`load_font`](https://github.com/Kolaru/MathTeXEngine.jl/blob/f07a5dfab247e9144dfc1076ab13a1badc14c156/src/engine/fonts.jl#L10-L23). A proper solution would require some changes to MathTeXEngine as well, I suppose, e.g. these font values should be allowed to be `Union{String,FTFont}` as returned by e.g. `FreeTypeAbstraction.findfont`. I suspect that Makie uses something similar. However, I can't find how/where the fonts are passed to MathTeXEngine.

Is this something you have on the radar? If not, could you give me some pointers, please? Thanks in advance! I won't be able to work on this within the next month, though. 🙂

CC @Kolaru

**Comment by jonas-schulze** [comment]
@SimonDanisch, did you have time to think about my comment above? If not, no rush. I'm just pinging you because I thought you are online, as you responded so quickly in #1296.

I don't know how LaTeX deals with superscripts exactly, but at least `pdflatex` doesn't use superscript glyphs (the font I use only has glyphs for 1, 2, 3; cf https://github.com/JuliaPlots/Makie.jl/issues/1296#issuecomment-1063867265). If MathTeXEngine does something similar, which I suppose as e.g. `L"K^{1234567890}_{1234567890}"` works fine (except the font being used), fixing this issue would be a work-around for #1296.

**Comment by SimonDanisch** [comment]
That's more of a question for @jkrumbiegel  and @Kolaru

**Comment by Kolaru** [comment]
The way consistency of fonts is currently supported is only the other way around: you can get the fonts used by MathTeXEngine using `get_font` or `get_fontpath` and ask Makie to use them (see https://github.com/Kolaru/MathTeXEngine.jl/pull/27).

The other way is not yet implemented, but now has its very own issue: https://github.com/Kolaru/MathTeXEngine.jl/issues/55

@jonas-schulze You seem to be pretty close. The main problem you may face is that the math font is (for Computer Modern) a special font with special encoding, that requires some additional data (namely how to fetch the special character).

If you get back to it before I can have a closer look (I will try to go through MathTeXEngine issues in the coming few weeks), please open an issue on MathTeXEngine.

**Comment by greimel** [comment]
When installed, you can use `fig = Figure(font = "CMU")`, then you will have a consistent font (Computer Modern) everywhere.

![image](https://user-images.githubusercontent.com/6280307/162177699-073c4b71-3eae-46ea-9424-5f2df1052097.png)

**Comment by jonas-schulze** [comment]
That is correct. But it doesn't help in situations where a user e.g. prefers the default font of Makie, or wants to have the figure's font (or all its fonts) match the non-CMU font used in the document the figure should be embedded in.

**Comment by greimel** [comment]
True, but it might be enough for OP.

**Comment by Kolaru** [comment]
I have added a simple support for TeXGyreHeros (Makie default) in MathTeXEngine, and a hack to use it: just put the command `\fontfamily{TeXGyreHeros}` somewhere in the LaTeXString (see https://github.com/Kolaru/MathTeXEngine.jl/pull/123).

Now this is still a hack, so maybe on the Makie side we would like to have a different interface to pass any FontFamily down to MathTeXEngine.

(there is of course also a lot of work to get any font to get a proper layout, including TeXGyreHeros, but that's a different story)
